### DISCUSSION TITLE
Anyone Running Artisan on Mac with macOS Sonoma?

### DISCUSSION THREAD

**Discussion by jwarthman** [artisan-roaster-scope/artisan#1274] [Q&A]
Before I update this laptop from macOS Ventura to Sonoma, I'd like to hear others' experiences running Artisan on Sonoma. Any issues?

**Comment by MAKOMO** [comment]
NO version of Artisan is currently compatible with macOS 14 Sonoma!

**Reply by MAKOMO** [reply]
I have to correct myself. The official release v2.8.4 seems to work under macOS 14 Sonoma (at least on Intel-based macs; still don't have access to any Apple Silicon). The current development builds seem to have problems.

**Reply by jwarthman** [reply]
Thank you, that's encouraging! I was disappointed at first, but your follow-up message is cause for hope.  :-)

**Reply by avimentor** [reply]
I'm successfully running the M1 Test 5 version on a MacBook M2 with Sonoma 14.0 connected to a Hottop KN-8828B-2K+.

https://github.com/artisan-roaster-scope/artisan/releases/tag/m1-test5

**Reply by shawnaks5** [reply]
@avimentor I tried this version (M1 Test 5) on my MacBook M1 with Sonoma 14.0 connected to a San Franciscan SF-6 and it does not work.  Still getting the Modbus fail error.

**Comment by RNCCoffee** [comment]
I have an Intel based Powerbook and a Mac Mini with an M2 chip. If I can update my Powerbook to Sonoma, I'll see if it works or not. Might take me a week or two to get around to it though...

**Comment by williamcramer** [comment]
I have a Macbook air M2 on Sonoma 14.1.1 running Artisan 2.8.4 (ac17fbe). All good, from what I can tell. Driving a TC4 arduino over usb serial, controlling a quest M3 via SSR's and an AC dimmer for the fan.  I have roasted 3 batches back to back every week for the last month or so, which takes about an hour to finish with ramp up, soak, roast cool down cycles. Very stable, no glitches.

**Reply by shawnaks5** [reply]
Great to hear!  I was finally able to get the latest version of Artisan to work with my M1 MacBook Pro on the Sonoma OS. I really don't know what exactly fixed the issue.   I saved my Artisan settings, uninstalled version 2.8.4, shut down and restarted the computer, reinstalled 2.8.4, and it started working.  I'm running a Watlow Modbus on a San Franciscan SF-6 with a SerialComm Isolated USB To RS485/RS422 Adapter.

**Comment by waterwynde** [comment]
I cannot get a new MacBook Air M3 running Sonoma 14.5 to connect to a 6-year old Hottop roaster.  My old MacBook Air running Monterrey 12.7 worked fine, but now that I've upgraded it to Sonoma 14.5 it no longer connects either.

I am able to select the port in Artisan, but there's no communication.  I authorized the new device on the USB port (and changed my Setting to Always Allow).

I assume the problem is Sonoma 14.5 is trying to negotiate with Hottop and that's failing so it does not enable... just a guess.

I am wondering about making a "data-only" USB cable with +5V cut ... I'm very reluctant to try this though.

Any thoughts?

**Reply by MAKOMO** [reply]
Do you use an USB dongle from Apple or a third party one. Third party USB dongles are known to make troubles on newer systems.

**Reply by avimentor** [reply]
Try an Apple-branded USB-C to USB-A adapter. I never got any third-party adapter to work with the HotTop.

I use a short, 1-foot USB cable that is more robust than the one that came with the HotTop. I added some clip-on ferrite coils to the USB cable to prevent electrical interference from an exhaust fan that was causing the HotTop to unexpectedly disconnect. I found the USB cable and ferrite coils on Amazon. 

With this set-up, everything has been rock solid.

**Comment by waterwynde** [comment]
Thanks for your question… using an Apple dongle that worked until I upgraded to Sonoma 14.5.

**Comment by waterwynde** [comment]
Thanks for your comments… I am using an Apple-branded dongle.  I can try the ferrite coils, but the fan is not running…

**Reply by williamcramer** [reply]
I have an external PSU on the TC4, so not driving the thing from the USB power. I do have issues with shared power on TC4 and Mac recognising the USB port. I need to follow a set power up sequence to have the USB port recognised, which I think is USB power, then enable the external power. Then the port is enabled and Artisan can connect, but I think it's a Mac issue, from what I can tell, and not Artisan. btw, I have the same with a generic USB dongle and another separate generic USB cable. Sonoma 14.5 M2 air.

I suggets you check this 1st:
https://apple.stackexchange.com/questions/245079/usb-debugging-identifying

I get an Arduino listed in system report when the TC4 is plugged in, but no idea what a HT will report. You should see something tho, e.g:
Product ID: 0x0043
Vendor ID: 0x2341

If you register as an Apple dev, you can get access to a probe util that might show more, but I've never had to go down that deep.
https://stackoverflow.com/questions/20674876/any-software-utility-available-to-do-usb-port-sniffing-in-mac-osx

**Comment by waterwynde** [comment]
Thanks for your thoughts...  I tried some different power-up / connect sequences and it didn't make a difference.

My MacBook Air M3 appears to see something when I connect the USB cable to a powered on Hottop, because the System Report for the USB branch shows the UART, whereas it shows nothing when the roaster is powered off.  I use "FT230X..." as the Comm Port in Hottop.  (Oddly, it first shows the Serial Number as the pre-selected port, but the drop-down only shows FT230X...
<img width="546" alt="Screenshot 2024-08-19 at 12 30 10 PM" src="https://github.com/user-attachments/assets/dad8a01a-1343-487c-ab9f-199213d22aa6">
